src/main/java/com/vinyl/service/EmailValidator.java

```diff
@@ -0,0 +1,28 @@
+package com.vinyl.service;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+import com.vinyl.model.User;
+import com.vinyl.service.exception.BadRequestException;
+
+@Component
+public class EmailValidator {
```

Comment:
both this class and the name validator are validators, they should be related

Reply:
done